src/actions/commands/actions.ts

```diff
@@ -2902,9 +2935,14 @@ class CommandGoToDefinition extends BaseCommand {
   isJump = true;
 
   public async exec(position: Position, vimState: VimState): Promise<VimState> {
-    await vscode.commands.executeCommand('editor.action.goToDeclaration');
+    const oldActiveEditor = vimState.editor;
 
-    if (vimState.editor === vscode.window.activeTextEditor) {
+    await vscode.commands.executeCommand('editor.action.goToDeclaration');
+    // `executeCommand` returns immediately before cursor is updated
+    // wait for the editor to update before updating the vim state
+    // https://github.com/VSCodeVim/Vim/issues/3277
+    await waitForCursorSync(1000);
```

Comment:
We can remove this hack. VSCode fixed the underlying issue.

Reply:
That probably came from a poor merge. Deleted.